lib/utils/net-utils.js

```diff
@@ -0,0 +1,25 @@
+const net = require('net')
+
+const NetUtils = {
+  isPortAvailable (port) {
+    return new Promise((resolve) => {
+      const server = net.createServer()
+
+      server.unref()
+      server.on('error', () => resolve(false))
+
+      server.listen({ port }, () => {
```

Comment:
We have to know whether a process is listening on a port, not just whether the port was available. These aren't exactly the same thing. I believe you also have to specify the host to prevent `EACCES`:
```js
 server.listen(port, '0.0.0.0', ()=>{...});
```

Reply:
Thanks, good point with host - I've updated pull request.

I think `isPortAvailable` function actually checks whether other process is listening on a port. `isPortAvailable` try to run server, if everything is ok it close server and resolves with true. 

It seems to be enough, but I may miss something.